sale_order_line_packaging_qty/models/sale_order_line.py

```diff
@@ -86,5 +87,60 @@ def _onchange_product_uom_qty(self):
         # TODO Drop once https://github.com/odoo/odoo/pull/49150/ is merged
         res = super()._onchange_product_uom_qty()
         if not res:
-            res = self._check_package()
+            res = self._check_package() or self._check_qty_is_pack_multiple()
         return res
+
+    @api.onchange("product_id")
+    def product_id_change(self):
+        res = super().product_id_change()
+        if self.product_id.sell_only_by_packaging:
+            self.product_uom_qty = min(self.product_id.packaging_ids.mapped("qty"))
+        return res
+
+    def _check_qty_is_pack_multiple(self):
+        """ Check only for product with sell_only_by_packaging
+        """
+        # and we dont want to have this warning when we had the product
+        if self.product_id.sell_only_by_packaging:
+            if not self._is_pack_multiple():
+                warning_msg = {
+                    "title": _("Product quantity can not be packed"),
+                    "message": _(
+                        "For the product {prod}\n"
+                        "The {qty} is not the multiple of any pack.\n"
+                        "Please add a package"
+                    ).format(prod=self.product_id.name, qty=self.product_uom_qty),
+                }
+                return {"warning": warning_msg}
+        return {}
+
+    def _is_pack_multiple(self):
+        return bool(self.product_id._which_pack_multiple(self.product_uom_qty))
+
+    def write(self, vals):
+        # Fill the packaging if they are empty and the quantity is a multiple
+        for line in self:
+            product_uom_qty = vals.get("product_uom_qty")
+            product_packaging = vals.get("product_packaging")
+            if line.product_id.sell_only_by_packaging and (
+                not line.product_packaging
+                or ("product_packaging" in vals and not product_packaging)
+            ):
+                pack_multiple = line.product_id._which_pack_multiple(product_uom_qty)
+                if pack_multiple:
+                    vals.update({"product_packaging": pack_multiple.id})
```

Comment:
How could this go unnoticed? :astonished: 

1. If `product_id` is defined in `vals` accessing `line.product_id` to select a packaging would use packaging of a product that is not the one written on the lines.
2. If `product_id` is not defined in `vals`, the for loop on the lines will redefine `product_packaging` on each iteration that finds a packaging, and the value written on all the lines will be the last packaging to be found.